Detectors/PHOS/reconstruction/src/AltroDecoder.cxx

```diff
@@ -65,8 +65,14 @@ void AltroDecoder::readChannels(const std::vector<uint32_t>& buffer, CaloRawFitt
       if (currentword != 0) {
         LOG(ERROR) << "Channel header mark not found, header word " << currentword;
         short fec = header.mHardwareAddress >> 7 & 0xf; //try to extract FEE number from header
-        if (fec < 0 || fec > 13) {
-          fec = kGeneralSRUErr;
+        short branch = header.mHardwareAddress >> 11 & 0x1;
+        if (branch < 0 || branch > 1) {
+          fec = kGeneralTRUErr + kGeneralSRUErr;
+        } else {
+          if (fec < 0 || fec > 13) {
+            fec = kGeneralSRUErr;
```

Comment:
also here, fec is limited to 0:15 ( = 0xf), so the <0 comparison is meaningless.

Reply:
Rudiment. Fixed in new version